src/mlpack/core/cv/k_fold_cv.hpp

```diff
@@ -0,0 +1,219 @@
+/**
+ * @file k_fold_cv.hpp
+ * @author Kirill Mishchenko
+ *
+ * k-fold cross-validation.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_CV_K_FOLD_CV_HPP
+#define MLPACK_CORE_CV_K_FOLD_CV_HPP
+
+#include <mlpack/core/cv/meta_info_extractor.hpp>
+#include <mlpack/core/cv/cv_base.hpp>
+
+namespace mlpack {
+namespace cv {
+
+/**
+ * The class KFoldCV implements k-fold cross-validation for regression and
+ * classification algorithms.
+ *
+ * To construct a KFoldCV object you need to pass the k parameter and arguments
+ * that specify data. For the latter see the CVBase constructors as a reference
+ * - the CVBase constructors take exactly the same arguments as ones that are
+ * supposed to be passed after the k parameter in the KFoldCV constructor.
```

Comment:
> and because we are not exposing the CVBase
> constructors directly to the user

What do you mean? The `CVBase` constructors are public.

Anyway, you have mentioned it already the second time (the first time was when you did it for the `SimpleCV` constructors). So, I guess it's pretty much important in general and to be consisted with other mlpack in particular - more important than I thought initially. In such a case I'm ready to do what you ask, even though I like the idea of avoiding code duplication. If we want to do it here, I guess it makes sense to do it in `SimpleCV` too.